### ISSUE TITLE
Writing a new model for Dell X-series, having some issues with timeouts.

### ISSUE THREAD

**Issue by o1crimson1o** [ytti/oxidized#794]
Seems to connect (using powerconnect) but then waits on input till it times out.

Seems related to the 

More: <space>,  Quit: q or CTRL+Z, One line: <return> 

Prompt. 

```
expect /^\s*--More--\s+.*$/ do |data, re|
     send ' '
     data.sub re, ''
  end
```

I think this is the line. 

I'm wondering should I look into extended the powerconnect or this enough of a justification to make a new model?

**Comment by crimson** [comment]
User Name:admin
Password:******


console#terminal datadump
% Unrecognized command
console#terminal length 0
% Unrecognized command
console#show version
SW version    3.0.0.80 ( date  19-May-2016 time  18:32:52 )
Boot version    1.0.0.17 ( date  19-Dec-2015 time  20:01:32 )
HW version    00.00.02
console#show system
System Description:                       Dell Networking X1008P 1Gb PoE Switch
System Type:                              X1008P
System Up Time (days,hour:min:sec):       07,00:24:41
System Contact:                           
System Name:                              
System Location:                          
System MAC Address:                       64:00:6a:e7:96:3d
System Object ID:                         1.3.6.1.4.1.674.10895.3068


console#show running-config
config-file-header
v3.0.0.80 / RASTUTE_800_026
CLI v1.0
set system mode L2 
           policy-based-vlans active 

@
port jumbo-frame
vlan database
vlan 1032,2000,2004
exit
voice vlan oui-table add 000181 Nortel__________________
voice vlan oui-table add 0001e3 Siemens_AG_phone________
voice vlan oui-table add 00036b Cisco_phone_____________
voice vlan oui-table add 00096e Avaya___________________
voice vlan oui-table add 000fe2 H3C_Aolynk______________
voice vlan oui-table add 001049 Shoretel________________
voice vlan oui-table add 0060b9 Philips_and_NEC_AG_phone
voice vlan oui-table add 00907a Polycom/Veritel_phone___
voice vlan oui-table add 00e0bb 3Com_phone______________
no eee enable
username admin password encrypted @#$@#$@#$@#$@#$@# privilege 15
snmp-server community Public ro view Default
More: <space>,  Quit: q or CTRL+Z, One line: <return>

**Comment by crimson** [comment]
expect /^\s*More:\s+.*$/ do |data, re|
    send ' '
    data.sub re, ''
  end

Matches the hit a key to continue crap. ( More: <space>,  Quit: q or CTRL+Z, One line: <return> )

I am still seeing it pause at that line, which leads me to believe I'm not using the proper send ' '
send ' ' from what I can tell is a space. Which should make it page.

**Comment by ytti** [comment]
Do you mean pause that it doesn't move forward at all? Or just that it takes while before it moves forward?

**Comment by crimson** [comment]
It doesn't move forward and times out. I've had a few Ruby dens I know look
at it, but they couldn't help me much.

**Comment by ytti** [comment]
How do you know that you're actually matching the expect? You could try to add some debugging there. Like ```Oxidized.logger.debug "saw pager prompt"```

**Comment by crimson** [comment]
Def check, but I used a ruby regex tester to very what I thought I was
doing. I will check and report back.

**Comment by WilliButz** [comment]
I got this working with our X-Series switches (models X1008 and X1026). As @ytti mentiond, the problem was that the displayed prompt is not just
```
More: <space>, Quit: q or CTRL+Z, One line: <return>
```
which would be matched by @crimson's regex
```
/^\s*More:\s+.*$/
```
But it seems like the prompt has some control characters at the beginning which I could so far match with
```
/^[[:cntrl:]]...More:\s+.*$/
```
but as of now I did not find a solution to get the escaped control characters printed with ruby to extend the regex properly to fit both powerconnect and X-Series switches.


(This should also fix #1198)

**Comment by jvkassi** [comment]
Hello @WilliButz,

Can you please share your model file  ?
